src/adapters/MorphoMarketV1Adapter.sol

```diff
@@ -106,6 +109,49 @@ contract MorphoMarketV1Adapter is IMorphoMarketV1Adapter {
         emit BurnShares(marketId, supplySharesBefore);
     }
 
+    function submitSetMovedSharesRecipient(address newMovedSharesRecipient) external {
+        require(msg.sender == IVaultV2(parentVault).curator(), NotAuthorized());
+        require(movedSharesRecipientExecutableAt == 0, AlreadyPending());
+        movedSharesRecipientExecutableAt = block.timestamp + IVaultV2(parentVault).timelock(IVaultV2.removeAdapter.selector);
+        emit SubmitSetMovedSharesRecipient(movedSharesRecipientExecutableAt);
+    }
+
+    function revokeSetMovedSharesRecipient() external {
+        require(msg.sender == IVaultV2(parentVault).curator() || IVaultV2(parentVault).isSentinel(msg.sender), NotAuthorized());
+        require(movedSharesRecipientExecutableAt != 0, NotPending());
+        movedSharesRecipientExecutableAt = 0;
+        emit RevokeSetMovedSharesRecipient();
+    }
+
+    function setMovedSharesRecipient(address newMovedSharesRecipient) external {
+        require(movedSharesRecipientExecutableAt != 0, NotPending());
+        require(block.timestamp >= movedSharesRecipientExecutableAt, TimelockNotExpired());
+        movedSharesRecipientExecutableAt = 0;
+        movedSharesRecipient = newMovedSharesRecipient;
+        emit SetMovedSharesRecipient(newMovedSharesRecipient);
+    }
+
+    function moveShares(bytes32 marketId) external {
+        require(movedSharesRecipient != address(0), MovedSharesRecipientNotSet());
+        require(blockNumberWhenMovedShares[marketId] == 0, AlreadyMoved());
+        blockNumberWhenMovedShares[marketId] = block.number;
+
+        MarketParams memory marketParams = IMorpho(morpho).idToMarketParams(Id.wrap(marketId));
+        uint256 supplyAssets = expectedSupplyAssets(marketId);
+        bytes memory data = abi.encode(marketParams);
+        IMorpho(morpho).supply(marketParams, supplyAssets, 0, address(this), data);
```

Comment:
This should go though most of the time, although:
- it's not completely trivial, so we could test/verify it
- it can leave some dust shares behind (which seems ok)

It can fail if there is not enough liquidity on Morpho (not on the market). This can probably be mitigated by a flashloan on another venue